### PR TITLE
Fix dialog resizing issues

### LINKED ISSUE #4511: Dialogs not returning to proper size after expanding
**Describe the bug**
After adding new items to a dialog, it no longer returns to the proper size when something is removed.

**To Reproduce**

1. Go to a dialog with a grid layout (OtherIdentifiers on https://umherb11024-edge.test.specifysystems.org/specify/view/collectionobject/new/)
3. Add new items
4. See it resize
6. See that it does not reduce in size when the items are removed

https://github.com/specify/specify7/assets/37256050/15be69ed-ebf9-445a-b9fd-4ea046145193

Arc, macOS

### LINKED ISSUE #4519: Cannot leave fullscreen in GeoMap
**Describe the bug**
Once you enter full screen in GeoMap, you can't escape.

**To Reproduce**

1. Run a query (https://umherb11024-edge.test.specifysystems.org/specify/query/fromtree/taxon/109665/)
2. Use the 'GeoMap' button to map all the points in the query
3. Use the full screen button
4. Try to leave full screen
5. See issue

https://github.com/specify/specify7/assets/37256050/8d49c6bf-aa9e-491b-a003-346d0e4f73ba

### PR DESCRIPTION
**Pull request by CarolineDenis**
Fixes #4519

Fixes #4511 

### Checklist

- [ ] Self-review the PR after opening it to make sure the changes look good
      and self-explanatory (or properly documented)
- [ ] Add relevant issue to release milestone

### Testing instructions

Code lost during merge

### CODE REVIEW

**Review by emenslin — APPROVED** [review]
- [x] Can now exit out of full screen in GeoMap
- [x] Dialogs now shrink back down in size when things are removed 

https://github.com/specify/specify7/assets/150940840/e98b6b5e-3bd2-4dff-b792-d184fc72c2ef

**Review by pashiav — COMMENTED** [review]
- [x] Can now exit out of full screen in GeoMap
- [x] Dialogs now shrink back down in size when things are removed

**Review by pashiav — APPROVED** [review]

File: specifyweb/frontend/js_src/lib/components/Molecules/Dialog.tsx [hunk 1]
```diff
@@ -606,44 +616,63 @@ function useTitleChangeNotice(dimensionKey: string | undefined): void {
   }, [dimensionKey]);
 }
 
+const freezeSizeAfter = 2 * SECOND;
```
Comment (maxpatiiuk) [inline specifyweb/frontend/js_src/lib/components/Molecules/Dialog.tsx:619]: When the code was copied, not all changes got copied - the original PR fixed the definition of `SECOND` which is not present in this PR

**Review by maxpatiiuk — CHANGES_REQUESTED** [review]
Let's merge the original PR (https://github.com/specify/specify7/pull/4535) instead of creating a duplicate PR

**Comment by maxpatiiuk** [comment]
> Code lost during merge

Code wasn't lost)
The PR just needs to be merged

As a reminder:
- I had a time-units-simplifications PR for refactoring time units
- The dialog fixes branch was based on it since it relied on the changes in that PR.
- Dialog fixes branch is merged (into time-units-simplifications)
- time-units-simplifications should be merged too to get the fixes into production

**Comment by CarolineDenis** [comment]
Fixed in https://github.com/specify/specify7/pull/4535

**Pull request closed without merging**